File: Challenger.Web/src/app/components/food/meal-add-record/MealRecordDto.ts
import { FastRecordDto } from "../fast-record-add/FastRecordDto";
import { MealDishDto } from "../meal-dish-add/MealDishDto";
import { MealProductDto } from "../meal-product-add/MealProductDto";

export class MealRecordDto {

    id: number;
    isNextDay: boolean;
    mealName: string;
    recordTime: number;
    diaryRecordId: number;
    mealProducts: MealProductDto[];
    mealDishes: MealDishDto[];
    fastRecords: FastRecordDto[];
}

File: Challenger.Domain/Dtos/MealRecordDto.cs
namespace Challenger.Domain.Dtos
{
    public class MealRecordDto
    {
        public long Id { get; set; }

        public bool IsNextDay { get; set; }
        
        public string MealName { get; set; }

        public int RecordTime { get; set; }

        public long DiaryRecordId { get; set; }

        public List<MealProductDto> MealProducts { get; set; }

        public List<MealDishDto> MealDishes { get; set; }

        public List<FastRecordDto> FastRecords { get; set; }
    }
}
